pkg/metrics/metrics.go

```diff
@@ -168,18 +168,17 @@ func (ms *metricsServer) buildupPendingInfo(ctx context.Context) {
 			return
 		case <-ticker.C:
 			totalPending := int64(0)
-			for _, pending := range ms.lagReaders {
-				if p, err := pending.Pending(ctx); err != nil {
+			for _, pendingLag := range ms.lagReaders {
+				if p, err := pendingLag.Pending(ctx); err != nil {
 					log.Errorw("Failed to get pending messages", zap.Error(err))
 				} else {
-					totalPending += p
+					if p != isb.PendingNotAvailable {
+						totalPending += p
+					}
```

Comment:
Under what circumstance can we have some of the pendingLag return pending information and others return PendingNotAvailable? And if that happens, why do we choose to calculate partial data instead of skipping the current ticker?

Reply:
since each partition is a separate stream, it's possible to have pendingNotAvailable for some of them.

Reply:
+1 to skip instead of sending partial information, will update the pr

Reply:
since it is a gauge does it really matter? there are 2 scenarios
* current pending gauge is > partial value (here sending a partial value will not help)
* current pending gauge is < partial value  (here sending a partial value will help)

isn't there a 50/50 chance for anything to happen? 😄

Reply:
Adding some context here to refresh my memory. `pending` is a metric used by UI to display and by controller to autoscale. For both case, I think we support calculating it use a user-specified look back seconds. We calculate `pending` of last n seconds by collecting last n-second `totalPending`s, adding them up, and then dividing them by n.

Yes, it's a gauge. If we only care about the current value, it's ok to send partial data, because if partial data is lower, we just don't scale up and otherwise, it could lead to a scale-up operation. However, we are looking back for n data points to calculate average, and under that case, I think accuracy become more important. We would rather collect n/2 accurate data points, sum and divide by n/2 than collect n data points with n/2 partial data points and divide by n.

Reply:
If we are storing the data and then dividing it, we shouldn't store partial data. thank you, Keran for the explanation.